src/skydb.ts

```diff
@@ -553,6 +653,23 @@ export function checkCachedDataLink(rawDataLink: string, cachedDataLink?: string
   return false;
 }
 
+/**
+ * Validates the given entry data.
+ *
+ * @param data - The entry data to validate.
+ * @throws - Will throw if the data is invalid.
+ */
+export function validateEntryData(data: Uint8Array): void {
```

Comment:
is it allowed to pass an empty `Uint8Array`? should it be?

Reply:
Do you mean a `null` array or an array of all zeros? Because passing an array of all zeros is how we implement delete, so if we disallow it here we'll need a different, internal, set method that allows it.

My 2c is to treat a zero array as any other value, including not returning `null` instead of it.

Reply:
That is a good point and I think that it should throw during runtime if you try to set the deletion sentinel. There is a good chance that someone doing that may not realize that it has a special meaning. And if they are trying to delete the entry, they should use the delete method instead.